app/src/main/java/com/github/libretube/util/DeArrowUtil.kt

```diff
@@ -0,0 +1,89 @@
+package com.github.libretube.util
+
+import android.util.Log
+import com.github.libretube.api.JsonHelper
+import com.github.libretube.api.RetrofitInstance
+import com.github.libretube.api.obj.ContentItem
+import com.github.libretube.api.obj.DeArrowContent
+import com.github.libretube.api.obj.StreamItem
+import com.github.libretube.constants.PreferenceKeys
+import com.github.libretube.extensions.toID
+import com.github.libretube.helpers.PreferenceHelper
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.decodeFromJsonElement
+
+object DeArrowUtil {
+    private fun extractTitleAndThumbnail(data: JsonElement): Pair<String?, String?> {
+        val content = try {
+            JsonHelper.json.decodeFromJsonElement<DeArrowContent>(data)
+        } catch (e: Exception) {
+            return null to null
+        }
+        val newTitle = content.titles.maxByOrNull { it.votes }?.title
+        val newThumbnail =
+            content.thumbnails.filter { it.thumbnail != null }.maxByOrNull { it.votes }
+                ?.takeIf { !it.original }?.thumbnail
+        return newTitle to newThumbnail
+    }
```

Comment:
Array is presorted optimally (manually approved submissions above others even if less votes). Just take the first element.

However, you still need to make sure the top value is valid, as it will return submissions at -1 votes.

```kt
        val newTitle = content.titles.firstOrNull?.takeIf { it.votes >= 0 || it.locked }.title
        val newThumbnail =
            content.thumbnails.filter { it.thumbnail != null }.firstOrNull?.takeIf { (it.votes >= 0 || it.locked) && !it.original }?.thumbnail
```

Reply:
I see, thanks!